src/Components/Card/Card.jsx

```diff
@@ -1,26 +1,29 @@
-import { Rating } from "./Rating";
-import { useData, useAuth } from "../../Helper";
 import { Link, useNavigate } from "react-router-dom";
+import { Rating } from "./Rating";
+import {
+  useAuth,
+  addToCart,
+  addToWishlist,
+  removeFromWishlist,
+} from "../../Helper";
 
 const Card = ({ product }) => {
-  const { isLoggedIn } = useAuth();
   const navigate = useNavigate();
-  const { inWishlist, inCart, dispatch } = useData();
-
-  const addToCartHandler = (product) => {
-    isLoggedIn
-      ? dispatch({ type: "ADD_TO_CART", payload: product })
+  const { userState, dispatchUserState, inWishlist, inCart } = useAuth();
+  const addToCartHandler = () => {
+    userState.token
+      ? addToCart(product, userState.token, dispatchUserState)
       : navigate("/login");
   };
 
-  const addToWishListHandler = (product) => {
-    isLoggedIn
-      ? dispatch({ type: "ADD_TO_WISHLIST", payload: product })
+  const addToWishListHandler = () => {
+    userState.token
+      ? addToWishlist(product, userState.token, dispatchUserState)
       : navigate("/login");
   };
 
-  const removeFromWishlistHandler = (product) => {
-    dispatch({ type: "REMOVE_FROM_WISHLIST", payload: product });
+  const removeFromWishlistHandler = () => {
+    removeFromWishlist(product, userState.token, dispatchUserState);
```

Comment:
`removeFromWishlist` method only needs `_id` from product, no need to pass the complete object here.
```suggestion
    removeFromWishlist(product._id, userState.token, dispatchUserState);
```

Reply:
fixed in https://github.com/saadahmadhere/elmart/pull/13/commits/1b5c9c8caec12d29427cb645b878d98bdb564977